subsys/net/ip/ipv6_nbr.c

```diff
@@ -2393,6 +2393,10 @@ static enum net_verdict handle_ra_input(struct net_pkt *pkt,
 
 	nd_opt_hdr = (struct net_icmpv6_nd_opt_hdr *)
 				net_pkt_get_data(pkt, &nd_access);
+
+	/* Add neighbor based on link local address received in Router Advertisement */
```

Comment:
This comment doesn't actually describe the line below. IMHO it would be worth mentioning that the neighbor cache entry is added regardless if the Router Advertisement contains a link-local address or not, because route entries reference data stored in the neighbor cache. If the link-local address is missing, it will be resolved using Neighbor Discovery protocol lazily, when trying to send a packet to the corresponding IPv6 address.

Reply:
Actually this comment is true, but I can extend it to mention link layer address. Link local address is always present, as it's just a source address of incoming Router Advertisement. Link layer address is the one that is added depending on the message content.